apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/local-simulation-benchmarking-and-gas-profiling.mdx

```diff
@@ -0,0 +1,251 @@
+import { FileTree } from "nextra/components";
+import { Callout } from 'nextra/components';
+
+# Local Simulation, Benchmarking & Gas Profiling
+
+## Overview
+The previous tutorial demonstrates how you can deploy and interact with Move contracts using various CLI commands.
+
+By default, those commands send a transaction to the remote fullnode for simulation and execution.
+You can override this behavior and simulate the transaction locally, by appending the following command line options of your preference:
+- `--local`: Simulate the transaction locally without additional alterations.
+- `--benchmark`: Benchmarck the transaction and report the running time(s).
+- `--profile-gas`: Profile the transaction for detailed gas usage.
+
+These additional options can be used in combination with the following CLI commands:
+- `aptos move run`
+- `aptos move run-script`
+- `aptos move publish`
+
+<Callout type="info" emoji="ℹ️">
+Local simulations do not result in any changes of the chain state.
+</Callout>
+
+## Deploying the Example Contract
+For demonstration purposes, we will continue to use the [`hello_blockchain`](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/hello_blockchain) package as an example.
```

Comment:
This doc talks about local simulation, benchmarking, etc. should we mention that it can be done without deployment to testnet/devnet (can it)?

Reply:
Added a note.